x-pack/plugins/security/public/management/roles/edit_role/privileges/kibana/feature_table/feature_table_expanded_row.test.tsx

```diff
@@ -351,4 +351,61 @@ describe('FeatureTableExpandedRow', () => {
     const object = wrapper.find('SubFeatureForm');
     expect(object.props()).toMatchObject({ disabled: false });
   });
+
+  it('require all spaces enabled and allSpacesSelected is true: option is enabled for both feature and sub-features', () => {
+    // Create role with `all` privilege turned on.
+    const role = createRole([
+      {
+        base: [],
+        feature: {
+          with_require_all_spaces_for_feature_and_sub_features: ['all'],
+        },
+        spaces: ['foo'],
+      },
+    ]);
+
+    const kibanaPrivileges = createKibanaPrivileges(kibanaFeatures);
+    const calculator = new PrivilegeFormCalculator(kibanaPrivileges, role);
+    const feature = kibanaPrivileges.getSecuredFeature(
+      'with_require_all_spaces_for_feature_and_sub_features'
+    );
+
+    // Make sure we update `selectedFeaturePrivileges` when `onChange` is called.
+    const onChange = jest.fn((_, selectedFeaturePrivileges) => {
+      wrapper.setProps({ selectedFeaturePrivileges });
+    });
```

Comment:
__note:__ this makes the test look more like an integration test, but I couldn't find another way to test the wrong behavior this PR fixes.